plugins/grafana-custom-plugins/grafana-dependency-plugin/README.md

```diff
@@ -0,0 +1,146 @@
+# Grafana Service Dependency Graph Plugin
+
+## What is Grafana Service Dependency Graph Plugin?
+
+Grafana Service Dependency Graph Plugin allows users to visualize how pods send
+and receive traffic to and from other pods and services. To help visualize the
+flows, the plugin groups pods by pod label and shows them 'inside' the node they
+are a part of.
+
+```mermaid
+graph LR;
+subgraph worker_node
+worker_pod_1
+worker_pod_2
+end;
+```
+A line pointing from source to destination is also shown, with the amount of
+data transmitted labelling the line.
+```mermaid
+graph LR;
+subgraph worker_node
+worker_pod_1 -- 8.4 MB --> worker_pod_2
+end;
+```
+## Acknowledgements
+
+The Service Dependency Graph Plugin is created using [mermaid-js](https://mermaid-js.github.io/mermaid/#/)
+
+## Data Source
+
+Supported Databases:
+
+- ClickHouse
+
+## Queries Convention
+
+Currently the Service Dependency Graph Plugin is created for restricted uses,
+only for visualizing transmitted data between pods and services. For correct
+loading of data for the Service Dependency Graph Plugin, the query is expected
+to return the following fields, in arbitrary order.
+
+- field 1: sourcePodName value with name or an alias of `sourcePodName`
+- field 2: sourcePodLabels value with name or an alias of `sourcePodLabels`
+- field 3: sourceNodeName value with name or an alias of `sourceNodeName`
+- field 4: destinationPodName value with name or an alias of `destinationPodName`
+- field 5: destinationPodLabels value with name or an alias of `destinationPodLabels`
+- field 6: destinationNodeName value with name or an alias of `destinationNodeName`
+- field 7: destinationServicePortName value with name or an alias of `destinationServicePortName`
+- field 8: octetDeltaCount value with name or an alias of `octetDeltaCount`
+
+Clickhouse query example:
+```sql
+SELECT sourcePodName,
+sourcePodLabels,
+sourceNodeName,
+destinationPodName,
+destinationPodLabels,
+destinationNodeName,
+destinationServicePortName,
+octetDeltaCount
+FROM flows
+WHERE ( destinationPodName IS NOT NULL AND destinationPodName != '' AND destinationPodName != 'undefined' )
+AND ( sourcePodName IS NOT NULL AND sourcePodName != '' AND sourcePodName != 'undefined' )
+AND ( positionCaseInsensitiveUTF8('${querySourcePodNamespace:raw}', sourcePodNamespace) > 0 )
```

Comment:
what's the reason for not simply do `sourcePodNamespace in (${querySourcePodNamespace:raw})`

Reply:
I tried that earlier, it returns one string with comma separated values, which 'in' is unable to use.

Reply:
I think I use it successfully in other dashboards under [snowflake](https://github.com/antrea-io/theia/tree/main/snowflake/grafana/provisioning/dashboards) directory, if you search `clusterUUID` in the dashboard JSON files, you'll be able to see. May be worth looking into what's the difference. It might be related to how you define the filter.

Reply:
Good idea, will look into it.